client.go

```diff
@@ -164,12 +148,30 @@ func (c *Client) Close() error {
 	return c.Text.Close()
 }
 
+func (c *Client) greet() error {
+	// Initial greeting timeout. RFC 5321 recommends 5 minutes.
+	c.conn.SetDeadline(time.Now().Add(5 * time.Minute))
```

Comment:
Let's make this configurable, e.g. `c.CommandTimeout` or `c.GreetingTimeout`.

Reply:
Good idea.